src/v/cluster/service.cc

```diff
@@ -87,6 +87,23 @@ service::create_topics(create_topics_request&& r, rpc::streaming_context&) {
       });
 }
 
+ss::future<create_topics_reply> service::create_non_replicable_topics(
+  create_non_replicable_topics_request&& r, rpc::streaming_context&) {
+    return ss::with_scheduling_group(
+             get_scheduling_group(),
+             [this, r = std::move(r)]() mutable {
+                 return _topics_frontend.local().create_non_replicable_topics(
+                   std::move(r.topics),
+                   model::timeout_clock::now() + r.timeout);
```

Comment:
Vadim has done some tests with this branch and has discovered timeouts returned from the `autocreate_non_replicable_topics` method. I suspected this line, so i created this test:
```

SEASTAR_THREAD_TEST_CASE(timer_test) {
    const auto max = seastar::lowres_clock_impl::steady_duration::max();
    const auto max_tp = seastar::lowres_clock_impl::steady_time_point::max();
    const auto now = seastar::lowres_clock_impl::steady_now();
    BOOST_CHECK((now + max) == max_tp);
}
```
and have noticed this fails, and furthermore a runtime error integer overflow is reported. I see you have copied this logic from `autocreate_topics` but it seems like it is incorrect there too and has only not created a problem probably because clients always pass a sane value for timeout when creating a topic.

In short, this should be fixed to work properly when a timeout of `no_timeout` is passed, and an issue will be filed to fix the existing issue in dev

Reply:
You can find failed test there: https://buildkite.com/vectorized/redpanda/builds/5123#8cbcb3f7-ce99-4a58-bd8f-8a03684aaae9

In logs for redpanda you can see
```
Exceptional future ignored: cluster::non_replicable_topic_creation_exception (Failed to disseminate non replicable topic: {ns: {kafka}, topic: {topic-yzdopvqkpj._second_topic_}}, error: cluster::errc:13
```
`error: cluster::errc:13` - is timeout

If put `no_timeout` in this line and in `dispatch_create_non_replicable_to_leader` - test works.
Also I try to print it 

```
auto t = model::timeout_clock::now() + timeout;
std::cout << t.time_since_epoch() << std::endl;
```

And got 
```
/home/vadim/redpanda/vbuild/llvm/install/bin/../include/c++/v1/chrono:1249:35: runtime error: signed integer overflow: 64417031 + 9223372036790358786 cannot be represented in type 'long long'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/vadim/redpanda/vbuild/llvm/install/bin/../include/c++/v1/chrono:1249:35 in 
-9223372036854775799
```

Reply:
https://github.com/vectorizedio/redpanda/issues/3184

Reply:
https://github.com/vectorizedio/redpanda/issues/3184